src/wallet/rpcwallet.cpp

```diff
@@ -3823,13 +3823,17 @@ RPCHelpMan getaddressinfo()
 
     LOCK(pwallet->cs_wallet);
 
-    UniValue ret(UniValue::VOBJ);
-    CTxDestination dest = DecodeDestination(request.params[0].get_str());
+    std::string error_msg;
+    CTxDestination dest = DecodeDestination(request.params[0].get_str(), error_msg);
+
     // Make sure the destination is valid
     if (!IsValidDestination(dest)) {
-        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid address");
+        if (error_msg.empty()) error_msg = "Invalid address";
```

Comment:
Maybe a comment why this can happen and why it's not caught in `DecodeDestination()`?

Reply:
I added an additional check in case someone changes the DecodeDestination method in the future and forgets to set the error_msg. I've added a comment explaining this.